assignment-client/src/scripts/EntityScriptServer.cpp

```diff
@@ -62,29 +62,38 @@ EntityScriptServer::EntityScriptServer(ReceivedMessage& message) : ThreadedAssig
 static const QString ENTITY_SCRIPT_SERVER_LOGGING_NAME = "entity-script-server";
 
 void EntityScriptServer::handleReloadEntityServerScriptPacket(QSharedPointer<ReceivedMessage> message, SharedNodePointer senderNode) {
-    auto entityID = QUuid::fromRfc4122(message->read(UUID_LENGTH_BYTES));
+    if (senderNode->getCanRez() || senderNode->getCanRezTmp()) {
+        auto entityID = QUuid::fromRfc4122(message->read(NUM_BYTES_RFC4122_UUID));
 
-    if (_entityViewer.getTree() && !_shuttingDown) {
-        qDebug() << "Reloading: " << entityID;
-        _entitiesScriptEngine->unloadEntityScript(entityID);
-        checkAndCallPreload(entityID, true);
+        if (_entityViewer.getTree() && !_shuttingDown) {
+            qDebug() << "Reloading: " << entityID;
+            _entitiesScriptEngine->unloadEntityScript(entityID);
+            checkAndCallPreload(entityID, true);
+        }
     }
 }
 
 void EntityScriptServer::handleEntityScriptGetStatusPacket(QSharedPointer<ReceivedMessage> message, SharedNodePointer senderNode) {
-    MessageID messageID;
-    message->readPrimitive(&messageID);
-    auto entityID = QUuid::fromRfc4122(message->read(UUID_LENGTH_BYTES));
-
-    // TODO(Huffman) Get Status
-    
-    qDebug() << "Getting script status of: " << entityID;
-    auto replyPacket = NLPacket::create(PacketType::EntityScriptGetStatusReply, -1, true);
-    replyPacket->writePrimitive(messageID);
-    replyPacket->writeString("running");
+    if (senderNode->getCanRez() || senderNode->getCanRezTmp()) {
```

Comment:
Yeah, this is a broader bug that should be fixed independently. It's also a very narrow edge case.